hadoop-ozone/cli-admin/src/main/java/org/apache/hadoop/hdds/scm/cli/container/ReconcileSubcommand.java

```diff
@@ -34,15 +45,183 @@
+        System.err.println("Failed to trigger reconciliation for container " + containerID + ": " + ex.getMessage());
+        failureCount++;
+      }
+    }
+    if (failureCount > 0) {
+      throw new RuntimeException("Failed trigger reconciliation for " + failureCount + " containers");
+    }
+  }
+
+  /**
+   * Used to json serialize the container and replica information for output.
+   */
+  private static class ContainerWrapper {
+    private final long containerID;
+    private final HddsProtos.LifeCycleState state;
+    private final ReplicationConfig replicationConfig;
+    private boolean replicasMatch;
+    private final List<ReplicaWrapper> replicas;
+
+    ContainerWrapper(ContainerInfo info, List<ContainerReplicaInfo> replicas) {
+      this.containerID = info.getContainerID();
+      this.state = info.getState();
+      this.replicationConfig = info.getReplicationConfig();
+
+      this.replicas = new ArrayList<>();
+      this.replicasMatch = true;
+      long firstChecksum = 0;
+      if (!replicas.isEmpty()) {
+        firstChecksum = replicas.get(0).getDataChecksum();
+      }
+      for (ContainerReplicaInfo replica: replicas) {
+        replicasMatch = replicasMatch && (firstChecksum == replica.getDataChecksum());
+        this.replicas.add(new ReplicaWrapper(replica));
+      }
+    }
+
+    public long getContainerID() {
+      return containerID;
+    }
+
+    public HddsProtos.LifeCycleState getState() {
+      return state;
+    }
+
+    public ReplicationConfig getReplicationConfig() {
+      return replicationConfig;
+    }
+
+    public boolean getReplicasMatch() {
+      return replicasMatch;
+    }
+
+    public List<ReplicaWrapper> getReplicas() {
+      return replicas;
+    }
+  }
+
+  private static class ReplicaWrapper {
+    private final DatanodeWrapper datanode;
+    private final String state;
+    private int replicaIndex;
+    @JsonSerialize(using = JsonUtils.ChecksumSerializer.class)
+    private final long dataChecksum;
+
+    ReplicaWrapper(ContainerReplicaInfo replica) {
+      this.datanode = new DatanodeWrapper(replica.getDatanodeDetails());
+      this.state = replica.getState();
+      // Only display replica index when it has a positive value for EC.
+      if (replica.getReplicaIndex() > 0) {
+        this.replicaIndex = replica.getReplicaIndex();
+      }
+      this.dataChecksum = replica.getDataChecksum();
+    }
+
+    public DatanodeWrapper getDatanode() {
+      return datanode;
+    }
+
+    public String getState() {
+      return state;
+    }
+
+    /**
+     * Replica index is only included in the output if it is non-zero, which will be the case for EC.
+     * For Ratis, avoid printing all zero replica indices to avoid confusion.
+     */
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    public int getReplicaIndex() {
+      return replicaIndex;
+    }
+
+    public long getDataChecksum() {
+      return dataChecksum;
+    }
+  }
+
+  private static class DatanodeWrapper {
```

Comment:
Do we need a DatanodeWrapper? Can we not just use DatanodeDetails#toString() ?

Reply:
It's harder to match json queries to the value returned by that method because the ID, hostname, and IP are all mixed together. If we want IP I can add it here.

Reply:
I've added IP address to the identifying information.